src/components/Dashboard.vue

```diff
@@ -273,130 +281,6 @@ export default Vue.extend({
-      const alternateSemesters = (catalogWhenOfferedDoesNotExist) ? [] : course.catalogWhenOffered!.replace(/\./g, '').split(', ');
-      const semesters = course.semesters || alternateSemesters;
-
-      // Get prereqs of course as string (). '' if neither available because '' is interpreted as false
-      const prereqs = course.prereqs || course.catalogPrereqCoreq || '';
-
-      // To be redefined if does not exist
-      // @ts-ignore
-      let { enrollment, lectureTimes, instructors }: {
-        enrollment: readonly string[];
-        lectureTimes: readonly string[];
-        instructors:readonly string[];
-      } = course;
-
-      if (!(enrollment || lectureTimes || instructors)) {
-        // If new course, iterate through enrollment groups to retrieve enrollment info, lecture times, and instructors
-
-        // Hash maps used to remove redundancies
-        const enrollmentMap: Record<string, boolean> = {};
-        const lectureTimesMap: Record<string, boolean> = {};
-        const instructorsMap: Record<string, string> = {};
-        course.enrollGroups.forEach(group => {
-          group.classSections.forEach(section => {
-            // Add section
-            const enroll = section.ssrComponent;
-            enrollmentMap[enroll] = true;
-
-            section.meetings.forEach(meeting => {
-              const { pattern, timeStart, timeEnd } = meeting;
-              // Only add the time if it is a lecture
-              if (enroll === 'LEC') lectureTimesMap[`${pattern} ${timeStart} - ${timeEnd}`] = true;
-
-              meeting.instructors.forEach(instructor => {
-                const { netid, firstName, lastName } = instructor;
-                instructorsMap[netid] = `${firstName} ${lastName}`;
-              });
-            });
-          });
-        });
-
-        enrollment = Object.keys(enrollmentMap);
-        lectureTimes = Object.keys(lectureTimesMap);
-        instructors = Object.keys(instructorsMap).map(netid => `${instructorsMap[netid]} (${netid})`);
-      }
-
-      // Distribution of course (e.g. MQR-AS)
-      // alternateDistributions option in case catalogDistr for the course is null, undef, ''
-      const catalogDistrDoesNotExist = (!course.catalogDistr) || course.catalogDistr === '';
-      const alternateDistributions = (catalogDistrDoesNotExist) ? [''] : /\(([^)]+)\)/.exec(course.catalogDistr!)![1].split(', ');
-      const distributions = course.distributions || alternateDistributions;
-
-      // Get last semester of available course. TODO: Remove when no longer firebase data dependant
-      const lastRoster = course.lastRoster || course.roster;
-
-      // Create course from saved color. Otherwise, create course from subject color group
-      const color = course.color || this.addColor(subject);
-
-      const alerts = { requirement: null, caution: null };
-
-      // Update requirements menu
-      this.updateRequirementsMenu();
```

Comment:
Previously we call this function to update the key for every class. I think it is unnecessary, and makes the refactoring harder. Therefore, I decide to call this function at the end of all updates. e.g. instead of updating requirement whenever we processed one class, we update them after processing all classes.